vaadin-flow-components-shared-parent/vaadin-flow-components-base/src/main/java/com/vaadin/flow/component/shared/ValidationUtil.java

```diff
@@ -28,72 +28,144 @@ private ValidationUtil() {
-     * @return <code>Validation.ok()</code> if the validation passes,
-     *         <code>Validation.error()</code> otherwise
+     * @param emptyValue
+     *            the empty value for the component
+     * @return {@code ValidationResult.ok()} if the value is not empty,
+     *         {@code ValidationResult.error()} otherwise
+     */
+    public static <V> ValidationResult checkRequired(String errorMessage,
+            boolean required, V value, V emptyValue) {
+        boolean isError = required && Objects.equals(emptyValue, value);
+        return isError ? ValidationResult.error(errorMessage)
+                : ValidationResult.ok();
+    }
+
+    /**
+     * Checks if the value satisfies the maximum value constraint and returns a
+     * {@code ValidationResult.ok()} or {@code ValidationResult.error()} with an
+     * empty error message depending on the result.
+     *
      * @param <V>
      *            the type of the component value
+     * @param value
+     *            the current value set on the component
+     * @param maxValue
+     *            the maximum value set on the component
+     * @return {@code ValidationResult.ok()} if the value is smaller or equal to
+     *         the maximum, {@code ValidationResult.error()} otherwise
+     * @deprecated since 24.5, use
+     *             {@link #checkMax(String, Comparable, Comparable)} instead.
      */
+    @Deprecated
     public static <V extends Comparable<V>> ValidationResult checkGreaterThanMax(
             V value, V maxValue) {
-        final boolean isGreaterThanMax = value != null && maxValue != null
-                && value.compareTo(maxValue) > 0;
-        if (isGreaterThanMax) {
-            return ValidationResult.error("");
-        }
-        return ValidationResult.ok();
+        return checkMax("", value, maxValue);
     }
 
     /**
-     * Checks if the value being set to the component is smaller than the max
-     * value defined
+     * Checks if the value satisfies the maximum value constraint and returns a
+     * {@code ValidationResult.ok()} or {@code ValidationResult.error()} with
+     * the given error message depending on the result.
      *
+     * @param <V>
+     *            the type of the component value
+     * @param errorMessage
+     *            the error message to return if the check fails
      * @param value
      *            the current value set on the component
-     * @param minValue
-     *            the min value set on the component
-     * @return <code>Validation.ok()</code> if the validation passes,
-     *         <code>Validation.error()</code> otherwise
+     * @param maxValue
+     *            the maximum value set on the component
+     * @return {@code ValidationResult.ok()} if the value is smaller or equal to
+     *         the maximum, {@code ValidationResult.error()} otherwise
+     */
+    public static <V extends Comparable<V>> ValidationResult checkMax(
```

Comment:
The new method names could be considered not descriptive enough. Some alternative options to consider:

- `validateMaxConstraint`
- `checkMaxConstraint`
- keep `checkGreaterThanMax`

Reply:
Thanks, good point. I like `validateMaxConstraint` the best.

Reply:
Renamed.